packages/flutter/test/material/tabs_test.dart

```diff
@@ -1758,6 +1758,68 @@ void main() {
     semantics.dispose();
   });
 
+  testWidgets('can be notified of TabBar onTap behavior', (WidgetTester tester) async {
```

Comment:
This test looks like it needs to be revised. It should confirm that the onTap() call is applied to the correct index and that it happens even when the default tap handler doesn't have any effect on the TabController.